src/object/C4Movement.cpp

```diff
@@ -184,122 +182,99 @@ int32_t C4Object::ContactCheck(int32_t iAtX, int32_t iAtY, uint32_t *border_hack
-			if (ctcox < lbound)
+			if (target_x < lbound)
 			{
-				StopAndContact(ctcox, lbound, xdir, CNAT_Left);
+				StopAndContact(target_x, lbound, xdir, CNAT_Left);
 			}
-			if (ctcox > rbound)
+			if (target_x > rbound)
 			{
-				StopAndContact(ctcox, rbound, xdir, CNAT_Right);
+				StopAndContact(target_x, rbound, xdir, CNAT_Right);
 			}
 		}
 	}
-	// landscape bounds
+	// Landscape bounds
 	C4Real lbound = itofix(0 - Shape.GetX());
 	C4Real rbound = itofix(::Landscape.GetWidth() - (Shape.GetX() + Shape.Wdt));
-	if (ctcox < lbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Sides))
+	if (target_x < lbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Sides))
 	{
-		StopAndContact(ctcox, lbound, xdir, CNAT_Left);
+		StopAndContact(target_x, lbound, xdir, CNAT_Left);
 	}
-	if (ctcox > rbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Sides))
+	if (target_x > rbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Sides))
 	{
-		StopAndContact(ctcox, rbound, xdir, CNAT_Right);
+		StopAndContact(target_x, rbound, xdir, CNAT_Right);
 	}
 }
 
-void C4Object::VerticalBounds(C4Real &ctcoy)
+void C4Object::VerticalBounds(C4Real &target_y)
 {
-	// layer bounds
+	// Layer bounds
 	if (Layer && (Layer->GetPropertyInt(P_BorderBound) & C4D_Border_Layer))
 	{
 		C4PropList* pActionDef = GetAction();
 		if (!pActionDef || pActionDef->GetPropertyP(P_Procedure) != DFA_ATTACH)
 		{
 			C4Real tbound = itofix(Layer->GetY() + Layer->Shape.GetY() - Shape.GetY());
 			C4Real bbound = itofix(Layer->GetY() + Layer->Shape.GetY() + Layer->Shape.Hgt - (Shape.GetY() + Shape.Hgt));
-			if (ctcoy < tbound)
+			if (target_y < tbound)
 			{
-				StopAndContact(ctcoy, tbound, ydir, CNAT_Top);
+				StopAndContact(target_y, tbound, ydir, CNAT_Top);
 			}
-			if (ctcoy > bbound)
+			if (target_y > bbound)
 			{
-				StopAndContact(ctcoy, bbound, ydir, CNAT_Bottom);
+				StopAndContact(target_y, bbound, ydir, CNAT_Bottom);
 			}
 		}
 	}
-	// landscape bounds
+	// Landscape bounds
 	C4Real tbound = itofix(0 - Shape.GetY());
 	C4Real bbound = itofix(::Landscape.GetHeight() - (Shape.GetY() + Shape.Hgt));
-	if (ctcoy < tbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Top))
+	if (target_y < tbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Top))
 	{
-		StopAndContact(ctcoy, tbound, ydir, CNAT_Top);
+		StopAndContact(target_y, tbound, ydir, CNAT_Top);
 	}
-	if (ctcoy > bbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Bottom))
+	if (target_y > bbound && (GetPropertyInt(P_BorderBound) & C4D_Border_Bottom))
 	{
-		StopAndContact(ctcoy, bbound, ydir, CNAT_Bottom);
+		StopAndContact(target_y, bbound, ydir, CNAT_Bottom);
 	}
 }
 
 void C4Object::DoMovement()
 {
-	int32_t iContact = 0;
-	bool fAnyContact = false;
-	int iContacts = 0;
-	BYTE fTurned = 0;
-	BYTE fRedirectYR = 0;
-	BYTE fNoAttach = 0;
+	bool has_moved = false;
+	bool has_contact = false;
+	int contact_count = 0;
+	BYTE has_turned = 0;
```

Comment:
Most legacy code uses integers for booleans because the old serialization code would only understand integers.

Check for all uses before turning it into bool though to make sure that the integer value isn't actually used.